internal/host/validator.go

```diff
@@ -841,3 +841,90 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) isAPIDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return ValidationSuccess
+	}
+	var response *models.DomainResolutionResponse
+	if err := json.Unmarshal([]byte(c.host.DomainNameResolutions), &response); err != nil {
+		v.log.WithError(err).Warn("Parse domain resolution")
+		return ValidationError
+	}
+
+	var hostsIPv4 []string
+	var hostsIPv6 []string
+
+	for _, h := range c.cluster.Hosts {
```

Comment:
What is this function trying to achieve?

- This function should only validate the API VIP domain, not other domains
- This function should validate it against the API VIP IP, not host interface IP addresses

Reply:
After some clarification the validation runs only for userMangedNetworks so the Api API domain is not defined, hence it should resolve api.{clusterName}.{baseDomain} and api-int.{clusterNmae}.{BaseDomain} to one of the hosts IPs in the cluster,  so this function checks the other hosts interfaces.
Everything else remains the same the only difference now is the resolved domains and the wanted IP.